### PR TITLE
Hover listener

### PR DESCRIPTION
**Pull request by atltt**
basic outline for this event listener class. Need some advice on just about everything

### CODE REVIEW

File: lib/hoverMain.js [hunk 1]
```diff
@@ -15,14 +21,17 @@
     activate (state) {
       this.subscriptions = new CompositeDisposable();
     },
-    this.subscriptions.add(atom.texteditor.getWordUnderCursor) {
+    this.subscriptions.add(atom.texteditor.getWordUnderCursor()) {
+      //add something here but unsure what that should be.
+      //Maybe add the console log for this.
```
Comment (robobenklein) [inline lib/hoverMain.js]: Yes - this looks like the hook function we want. Go ahead and try some logging here.

Comment (robobenklein) [reply lib/hoverMain.js]: Note: you will need this to get hooked by making sure the Atom plugin loader executes this as an `onload`

Comment (atltt) [reply lib/hoverMain.js]: > Note: you will need this to get hooked by making sure the Atom plugin loader executes this as an `onload`

Also shouldn't this be handled by the state activation to force itself into activation?

**Comment by robobenklein** [comment]
Let me know as soon as you start seeing output - remember Atom is an electron app and you can open the inspector with Ctrl-Shift-I to see the logging.

**Comment by atltt** [comment]
@robobenklein I'm having trouble loading it at all. It's non-existent at this point. Is it loading is it not serializing? I'm not sure what is going on with it.

**Comment by atltt** [comment]
This branch is looking GOOD

File: lib/hoverMain.js [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+  //}
+
+
+
+  //This looks promising, you can hover over a regex, eg "#5" and this
+  //will catch that and show it.
+
+ import {CompositeDisposable} from 'atom';
+
+  const cursorListener = {
```
Comment (robobenklein) [inline lib/hoverMain.js]: this indentation is awful

File: lib/hoverMain.js [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    deactivate () {
+      this.subscriptions.dispose();
+      this.cursorReturn.destroy();
+    },
+    serialize () {
+      //This shouldn't be needed in the actual plugin.
+    }
+  };
+
+  export default cursorListener;
```
Comment (robobenklein) [inline lib/hoverMain.js]: where did you actually attach the listener?

File: lib/hoverMain.js [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+  const cursorListener = {
+  subscriptions: null,
+  cursorReturn: null,
+    activate (state) {
+      this.subscriptions = new CompositeDisposable();
+    },
+
+    //This should be the main subscription we need.
+    //Others can be added into it to get gather other events.
+    this.subscriptions.add(atom.texteditor.getWordUnderCursor()) {
```
Comment (robobenklein) [inline lib/hoverMain.js]: this should be run with the onload to connect the event

File: lib/hoverMain.js [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+
+      //This should be the most we need to do.
+      var check = atom.texteditor.getWordUnderCursor();
+
+      //Here we should call a function to actually compare the result
+      //to our list of issue callsigns eg. "#13" or "Issue 14"
+
+      //Delete this later.
+      console.log(var);
```
Comment (robobenklein) [inline lib/hoverMain.js]: this is in the callback function or not?

File: lib/hoverMain.js (hunk 2 above)
Comment (atltt) [reply lib/hoverMain.js]: gg=G miene dude

File: lib/hoverMain.js (hunk 3 above)
Comment (atltt) [reply lib/hoverMain.js]: added onto package

File: lib/hoverMain.js (hunk 5 above)
Comment (atltt) [reply lib/hoverMain.js]: it is not

Comment (robobenklein) [reply lib/hoverMain.js]: but it should be!

File: lib/hoverMain.js (hunk 4 above)
Comment (atltt) [reply lib/hoverMain.js]: added into main ts func

Comment (atltt) [reply lib/hoverMain.js]: is it something like atom.onload in the main ts?

or is it the hoverMain.onload in the main ts?

**Comment by atltt** [comment]
FYI I don't have nodeJS on my laptop, keeps failing when trying to install

**Comment by atltt** [comment]
Some say this branch will never be merged

File: lib/hoverMain.js (hunk 4 above)
Comment (atltt) [reply lib/hoverMain.js]: I put it in the main ts

**Comment by atltt** [comment]
> More code to steal: https://github.com/ranjitjhala/hover-tooltips

Not this guy again.

**Comment by ilumsden** [comment]
@atltt take a look at this: https://github.com/TypeStrong/atom-typescript/blob/cfbc8306ce1baa71203b99f509bce7b6ac387f0b/lib/main/atom/tooltipManager.ts

The `TooltipManager` class is what you'd want to really look at.

**Comment by ilumsden** [comment]
It's not going to be super helpful until you convert your stuff to TypeScript, but it does show some stuff.

**Comment by ilumsden** [comment]
Also take a look at the `getMessage` function. It extracts the current word from the code.

**Comment by ilumsden** [comment]
@atltt @robobenklein @RexRascerX it might be worthwhile to take a look at this too:
https://github.com/atom-haskell/ide-haskell/search?q=tooltip&unscoped_q=tooltip

It's apparently where atom-typescript got its inspiration for the "show type on hover" tooltip, and it's another Atom plugin written at least partially in TypeScript.

**Comment by ilumsden** [comment]
My bad, wrong link. Here's the right one: https://github.com/atom-haskell/ide-haskell

**Comment by ilumsden** [comment]
The `src` directory has all the TypeScript code.

**Comment by atltt** [comment]
Nothing works!

**Comment by robobenklein** [comment]
https://atom.io/docs/api/v1.31.2/TooltipManager#instance-add

I just need to figure out how to extract the `HTMLElement` from the `TextBuffer` or `TextEditor` objects...

**Comment by robobenklein** [comment]
Also we should place markers at every issuetag we find: https://atom.io/docs/api/v1.31.2/TextEditor#instance-decorateMarker

**Comment by robobenklein** [comment]
Ok, I think I have an idea.

We can create `Decorations` from `Markers`: https://atom.io/docs/api/v1.31.2/TextEditor#instance-decorateMarker

then in that decoration we have the option to add a div with our custom class:

> highlight: Adds a new highlight div to the editor surrounding the range marked by the marker. When the user selects text, the selection is visualized with a highlight decoration internally. The structure of this highlight will be `html <div class="highlight <your-class>"> <!-- Will be one region for each row in the range. Spans 2 lines? There will be 2 regions. --> <div class="region"></div> </div>`
overlay: Positions the view associated with the given item at the head or tail of the given DisplayMarker.
block: Positions the view associated with the given item before or after the row of the given TextEditorMarker.

then we should be able to use that div with class to find the html element and apply a `Tooltip` on that

**Comment by atltt** [comment]
Add a right click sub event? No problem! If only I need how to do it!

**Comment by robobenklein** [comment]
image

Is this dead code now?

**Comment by atltt** [comment]
Meme Branch Please Ignore

**Comment by atltt** [comment]
It is with a heavy heart that I am closing this Pull Request. We remember the life of this Pull Request that led to this branch dying a lonely and hopeful life

**Pull request closed without merging**